### PR TITLE
Blaze: add new method to remotely check for eligibility

### PR DESCRIPTION
**Pull request by jeherve**
## Proposed changes:

This will allow us to remotely check eligibility for a site, and potentially update rules without having to release a new version of the package.

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion

* 1386-gh-dotcom-forge

## Does this pull request change what data or activity we track or use?

* No

## Testing instructions:

* In wp-admin, load the Posts list.
* You should see no "Blaze" post action link.
* In your WordPress.com sandbox, remove the check for Jetpack sites in the endpoint (see D98748-code to see how it can be done)
* The "Blaze" post action link should now appear.

### CODE REVIEW

File: projects/packages/blaze/src/class-blaze.php [hunk 1]
```diff
@@ -56,32 +56,66 @@ public function register() {
...
-	 * Determines if criteria is met to enable Blaze features.
+	 * Check the WordPress.com REST API
+	 * to ensure that the site supports the Blaze feature.
+	 *
+	 * @param int $blog_id The blog ID to check.
 	 *
-	 * @todo - Get response from API if requirements are met on the wpcom-side.
+	 * @return bool
+	 */
+	private static function site_supports_blaze( $blog_id ) {
```
Comment (dereksmart) [inline projects/packages/blaze/src/class-blaze.php]: Why private?

Comment (jeherve) [reply projects/packages/blaze/src/class-blaze.php]: Good point. Done in 347c4a42f0718bbea00fbe2007ec33d4c04c5917

**Review by DustyReagan — APPROVED** [review]
Tested on a Simple and Atomic site and the eligibility switch works! I also loaded this PR in Jurassic.Ninja and noted that Blaze was not enabled. 🎉

**Review by coder-karen — APPROVED** [review]
Tests well to me 👍

**Pull request merged**